pallets/router/src/benchmarking.rs

```diff
@@ -49,37 +68,54 @@ fn initial_set_up<T: Config<I>, I: 'static>(caller: T::AccountId) {
     )
     .ok();
 
-    <T as crate::Config<I>>::Assets::mint_into(DOT, &caller, INITIAL_AMOUNT.into()).ok();
+    <T as crate::Config<I>>::Assets::mint_into(DOT.into(), &caller, INITIAL_AMOUNT.into()).ok();
 
     let pool_creator = account("pool_creator", 1, 0);
-    <T as crate::Config<I>>::Assets::mint_into(DOT, &pool_creator, INITIAL_AMOUNT.into()).ok();
-    <T as crate::Config<I>>::Assets::mint_into(XDOT, &pool_creator, INITIAL_AMOUNT.into()).ok();
+    <T as crate::Config<I>>::Assets::mint_into(DOT.into(), &pool_creator, INITIAL_AMOUNT.into())
+        .ok();
+    <T as crate::Config<I>>::Assets::mint_into(XDOT.into(), &pool_creator, INITIAL_AMOUNT.into())
+        .ok();
 
     assert_ok!(pallet_amm::Pallet::<T>::add_liquidity(
         SystemOrigin::Signed(pool_creator).into(),
-        (DOT, XDOT),
-        (100_000_000, 100_000_000),
-        (99_999, 99_999),
-        10
+        (DOT.into(), XDOT.into()),
+        (100_000_000u128.into(), 100_000_000u128.into()),
+        (99_999.into(), 99_999.into()),
+        ASSET_ID.into()
     ));
 }
 
 benchmarks_instance_pallet! {
+     where_clause {
+        where
+            BalanceOf<T, I>: FixedPointOperand,
+            AssetIdOf<T, I>: AtLeast32BitUnsigned,
+            <<T as crate::Config<I>>::Assets as Inspect<T::AccountId>>::Balance: From<u128>,
+            <<T as crate::Config<I>>::Assets as Inspect<T::AccountId>>::AssetId: From<u32>,
+            <<T as pallet_amm::Config>::Assets as Inspect<<T as frame_system::Config>::AccountId>>::Balance:
```

Comment:
It's too complex here. Do we have to choose to inherit pallet-amm Config on this pallet?

Reply:
@yz89 I have to add this to fix compilation error. Regarding Router Pallet implementation, you may ask @ETeissonniere or @rustaceanstayhigh